vendor/k8s.io/gengo/args/args.go

```diff
@@ -103,6 +108,12 @@ func (g *GeneratorArgs) LoadGoBoilerplate() ([]byte, error) {
 		return nil, err
 	}
 	b = bytes.Replace(b, []byte("YEAR"), []byte(strconv.Itoa(time.Now().Year())), -1)
+
+	if g.GeneratedByComment {
+		cmdLine := strings.Join(os.Args, " ")
```

Comment:
Does this necessarily preserve the exact command that was run?
I tested this code
```
fmt.Println(strings.Join(os.Args, " "))
```
with command line arguments of
```
go run main.go --test="hello world"
```
And instead of printing
```
... --test="hello world"
```
It printed
```
... --test=hello world
```
This might not be an issue for the use cases we have, but I just thought I would mention it. I know we have --test_args="--arg1 --arg2 ..." but maybe our code generators never use this format?

Reply:
I'm just imagining a scenario where we intend to pass a flag to a subcommand but it ends up getting passed to the top level command, which could be bad especially if the flag is defined for both

Reply:
I don't see an easy fix for this either. We could parse the args and check if they contain spaces and if they do put quotes around just the value part, but that seems like a lot of work that might have been done before

Reply:
I looked briefly for a library doing shell escaping. So yes, we need something better.